Samplero/Samplero/Screen/Camera/CameraViewController.swift

```diff
@@ -0,0 +1,260 @@
+        button.layer.cornerRadius = UIScreen.main.bounds.width / 16.25
+        button.layer.masksToBounds = true
+        button.backgroundColor = .init(white: 1, alpha: 0.2)
+        button.setImage(UIImage(named: "cart"), for: .normal)
+        return button
+    }()
+    
+    // MARK: - Life Cycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        checkCameraPermissions()
+        addTargets()
+    }
+    
+    override func render() {
+        view.layer.addSublayer(previewLayer)
+        previewLayer.frame = view.bounds
+        
+        view.addSubview(topDrawer)
+        topDrawer.snp.makeConstraints { make in
+            make.top.leading.equalToSuperview()
+            make.width.equalToSuperview()
+            make.height.equalTo(UIScreen.main.bounds.height / 6.975)
+        }
+        
+        view.addSubview(bottomDrawer)
+        bottomDrawer.snp.makeConstraints { make in
+            make.bottom.leading.equalToSuperview()
+            make.width.equalToSuperview()
+            make.height.equalTo(UIScreen.main.bounds.height / 4.157)
+        }
+        
+        view.addSubview(shutterButton)
+        shutterButton.snp.makeConstraints { make in
+            make.bottom.equalToSuperview().inset(UIScreen.main.bounds.height / 10.55)
+            make.centerX.equalToSuperview()
+            make.size.equalTo(UIScreen.main.bounds.height / 14.07)
+        }
+        
+        view.addSubview(bringPhotoButton)
+        bringPhotoButton.snp.makeConstraints { make in
+            make.top.equalToSuperview().inset(UIScreen.main.bounds.height / 12.98)
+            make.leading.equalToSuperview().inset(UIScreen.main.bounds.width / 16.25)
+        }
+        
+        view.addSubview(photoHistoryButton)
+        photoHistoryButton.snp.makeConstraints { make in
+            make.leading.equalToSuperview().inset(UIScreen.main.bounds.width / 19.5)
+            make.centerY.equalTo(shutterButton)
+            make.size.equalTo(UIScreen.main.bounds.width / 7.96)
+        }
+        
+        view.addSubview(cartButton)
+        cartButton.snp.makeConstraints { make in
+            make.trailing.equalToSuperview().inset(UIScreen.main.bounds.width / 19.5)
+            make.centerY.equalTo(shutterButton)
+            make.size.equalTo(UIScreen.main.bounds.width / 8.125)
+        }
+    }
+    
+    override func configUI() {
+        #if targetEnvironment(simulator)
+        let image = UIImage(named: "sample_photo")
+        
+        session?.stopRunning()
+        
+        let imageView = UIImageView(image: image)
+        imageView.contentMode = .scaleAspectFill
+        imageView.frame = view.bounds
+        imageView.layer.zPosition = -1
+        view.addSubview(imageView)
+        #endif
+        view.backgroundColor = .black
+    }
+    
+    // MARK: - Func
+    
+    func addTargets() {
+        shutterButton.addTarget(self, action: #selector(didTapTakePhoto), for: .touchUpInside)
```

Comment:
p3.
```suggestion
        shutterButton.rx.tap.bind {
            print("clicked take photo button")

            #if !targetEnvironment(simulator)
            self.output.capturePhoto(with: AVCapturePhotoSettings(),
                                delegate: self)
            #endif
        }.disposed(by: disposeBag)
```

향후 Rx 코드로 전환하면 좋을 것 같습니다!!

Reply:
근데 여기도 `weak self` 써주어야 할까요? `weak self` 쓰면 169번쨰줄에 delegate로 넘겨줄때 self가 옵셔널인데 `self!` 처럼 강제 언래핑을 해야할지..

Reply:
> 근데 여기도 `weak self` 써주어야 할까요? `weak self` 쓰면 169번쨰줄에 delegate로 넘겨줄때 self가 옵셔널인데 `self!` 처럼 강제 언래핑을 해야할지..

https://jinsangjin.tistory.com/129 
`guard  let strongSelf = self else {return}`을 사용해도 좋을 거 같아요.
rx적으로는 `shutterButton.rx.tap. withUnretained(self).bind {~` !

Reply:
일단 rx로 고치고 추후 얘기해보는걸로 해요!